apps/server/src/migrations/migrations.ts

```diff
@@ -6,6 +6,66 @@
 
 // Migrations should be kept in descending order, so the latest migration is first.
 const MIGRATIONS: (SqlMigration | JsMigration)[] = [
+    // Add OCR results table for storing extracted text from images
+    {
+        version: 233,
+        sql: /*sql*/`\
+            -- Create OCR results table to store extracted text from images
+            CREATE TABLE IF NOT EXISTS ocr_results (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                entity_id TEXT NOT NULL,
+                entity_type TEXT NOT NULL DEFAULT 'note',
+                extracted_text TEXT NOT NULL,
+                confidence REAL NOT NULL,
+                language TEXT NOT NULL DEFAULT 'eng',
+                extracted_at TEXT NOT NULL,
+                created_at TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                updated_at TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                UNIQUE(entity_id, entity_type)
+            );
```

Comment:
Instead of storing the results in a separate table (that is not synced yet), how about simply storing the results in an attachment?

This:

* Simplifies the table structure
* Can easily filter for them by having a custom `role`.
* Allows additional functionality such as being able to view the attachment in order to review how well the OCR went.
* We can use the same `blobs` structure that is shared with notes and attachments.

The only problem is with image-attachments, since we can't have attachments for attachments.

Reply:
You definitely have the "whole picture" in mind - so do you think it's best to bite the bullet and create some new table/object that can be used/reused for objects (or even a more obscure "object", whatever that may be in the future) such as these, or do we just use an `attachment` for now?